core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -177,11 +179,15 @@ protected final SdkClientConfiguration syncClientConfiguration() {
         configuration = mergeChildDefaults(configuration);
         configuration = mergeGlobalDefaults(configuration);
 
+        // Invoke the plugins after defaults and before finalizing the configuration.
+        configuration = invokePluginsIfNeeded(configuration);
```

Comment:
Where was it invoked before this, and which `finalizing hooks` did you refer to in the PR description?

Reply:
I have updated the description to add more context and links to the referred code. Please let me know if that helps or if you would like me to expand on anything.

Reply:
> Where was it invoked before this,

This was not invoked before, now it is at client build time. This is the last step that is used to build final configuration for the client. Some of the supporting code was added before but this just now enabled.

Reply:
Ah, I see - the finalizing config methods were what you were referring to. Thank you for updating the PR description. Yes, we'd have to call it no later than here.